parser/src/context.rs

```diff
@@ -98,51 +135,54 @@ impl MathContext {
         // Trigonometric functions
         context.add_function(
             vec![Token::Backslash, Token::Identifier("sin".to_string())],
-            MathFunction::from_fn_pointer(|args, _| {
-                args[0].map_expecting_scalar(|v| v.sin())
-            }),
+            f64::sin,
```

Comment:
What is this? this looks absolutely bonkers...

Reply:
IntoMathFunction is implemented for all functions that have the signature fn(f64)->f64